application/helpers/pdf_helper.php

```diff
@@ -70,23 +71,28 @@ function generate_invoice_pdf($invoice_id, $stream = true, $invoice_template = n
         $custom_fields['quote'] = $CI->mdl_custom_fields->get_values_for_fields('mdl_quote_custom', $invoice->quote_id);
     }
 
-    // PDF associated files
-    $include_zugferd = $CI->mdl_settings->setting('include_zugferd');
-
-    if ($include_zugferd) {
-        $CI->load->helper('zugferd');
-
-        $associatedFiles = array(
-            array(
-                'name' => 'ZUGFeRD-invoice.xml',
-                'description' => 'ZUGFeRD Invoice',
-                'AFRelationship' => 'Alternative',
-                'mime' => 'text/xml',
-                'path' => generate_invoice_zugferd_xml_temp_file($invoice, $items)
-            )
-        );
-    } else {
-        $associatedFiles = null;
+    // START UBL+ changements
+    // Filename
+    $replace = array('.', ' ', '/', '-', '_', '\\', '#');                              
+    $filename = str_replace($replace, '', $invoice->user_vat_id) . '_' . str_replace($replace, '', $invoice->invoice_number);  
```

Comment:
You use the user vat id without checking if it is set. It's not a required field!